### PR TITLE
Adds repeated message squashing to chat.

### PR DESCRIPTION
**Pull request by AnturK**
:cl:
add: Repeated messages will now be squashed in the chat.
/:cl:

- [x] Don't repeat same operations & clean css

Majority of work done by @vuonojenmustaturska 

dreamseeker_2017-12-16_23-39-39

### CODE REVIEW

**Comment by Potato-Masher** [comment]
Aw heck, it applies to EVERYTHING, not just the normally repetitive messages? How will one get the effect of someone yelling the same thing over the radio if it doesn't flood your chat?

**Comment by Xhuis** [comment]
Woo! This is a really good change.

**Comment by vuonojenmustaturska** [comment]
00:50 < Naksu> Antur: you might want to remove the border radius etc from the css, they were there for my cute idea of having the span be a badge the same color as the text, with a white background
00:50 < Naksu> er
00:50 < Naksu> white text
00:50 < Naksu> on text-color background
00:51 < Naksu> lots of the other css is also redundant

reposting this here as a reminder

**Comment by praisenarsie** [comment]
just want to show this is how goon does their repeated messages, can we go something similar to theirs?

**Comment by vuonojenmustaturska** [comment]
that's easy, but it would also be easy to make it "pulse" the newest message

**Comment by Potato-Masher** [comment]
I would prefer it if only stuff that gets repeated often (feedback messages from game when your head gets beat in x10) rather than everything including IC and OCC.

**Comment by epochayur** [comment]
PR of the century

**Comment by SgtxRootbeer** [comment]
does this respect order of messages, ie
someone whacks me with a bucket twice,
then someone says shit over the radio
then back to bucket whacking

comes out as:
you were hit with a bucket! [x2]
urist mcbuckets says: "you're a fag"
you were hit with a bucket! [x4]

not as:
you were hit with a bucket! [x5]
urist mcbuckets says: "you're a fag"

**Comment by AnturK** [comment]
It only applies to consecutive messages. Two lines repeating won't get squashed.

The chat has no knowledge of message meaning, it's possible to define squashable class so only the content with that will get collapsed but i'm not sure that's the best idea.

**Comment by ShizCalev** [comment]
Not a fan of straight up increases the text size. Do love the pulsing vuono suggested though. 

Think a combination of pulsing the text and appending the multiplier to it would look very nice imo.

File: code/modules/goonchat/browserassets/js/browserOutput.js [hunk 1]
```diff
@@ -606,7 +636,15 @@ $(function() {
...
+	} 
+	if (savedConfig.smessagecombining) {
+		if (savedConfig.smessagecombining == 'false') {
+			opts.messageCombining = false;
+		} else {
+			opts.messageCombining = true;
+		}
+	}
+	else {
 		$('#adminMusic').prop('volume', opts.defaultMusicVolume / 100);
```
Comment (Tayyyyyyy) [inline code/modules/goonchat/browserassets/js/browserOutput.js]: You added a bug here by tying admin music volume to `savedConfig.smessagecombining`.

Also, `opts.messageCombining = !(savedConfig.smessagecombining === 'false')` would save you a few lines of unneeded code above.

Comment (AnturK) [reply code/modules/goonchat/browserassets/js/browserOutput.js]: Good catch, will fix this in a moment.

**Pull request merged**